src/Storages/ExternalStream/Kafka/Consumer.h

```diff
@@ -30,17 +31,24 @@ class Consumer : boost::noncopyable
 
     void consumeBatch(Topic & topic, Int32 partition, uint32_t count, int32_t timeout_ms, Callback callback, ErrorCallback error_callback) const;
 
-    void shutdown() { stopped.test_and_set(); }
+    void setStopped() {
+        stopped.test_and_set();
+        LOG_INFO(logger, "Stopped");
+    }
+
+    bool isStopped() const { return stopped.test(); }
 
     std::string name() const { return rd_kafka_name(rk.get()); }
 
 private:
-    void backgroundPoll(UInt64 poll_timeout_ms) const;
+    void backgroundPoll() const;
 
+    UInt64 poll_timeout_ms {0};
     klog::KafkaPtr rk {nullptr, rd_kafka_destroy};
     ThreadPool poller;
     Poco::Logger * logger;
 
+    std::atomic_flag started = ATOMIC_FLAG_INIT;
```

Comment:
Since c++20, we don't need explicitly init std::atomic_flag with `ATOMIC_FLAG_INIT` anymore since it is the default already